apps/prairielearn/elements/pl-file-upload/pl-file-upload.js

```diff
@@ -182,6 +255,30 @@
       return file ? file.contents : null;
     }
 
+    deleteUploadedFile(name) {
+      this.pendingFileDownloads.delete(name);
+      this.failedFileDownloads.delete(name);
+      var idx = this.files.findIndex((file) => file.name === name);
+      if (idx !== -1) {
+        this.files.splice(idx, 1);
+      }
+
+      // Recompute which regexes are matched by remaining files
+      this.requiredFilesUnmatchedRegex = this.requiredFilesRegex.slice();
+      this.files.forEach((n) => {
+        if (this.requiredFiles.includes(n.name)) return;
+        const matchingRegex = this.requiredFilesUnmatchedRegex.findIndex((f) =>
+          new RegExp(f[0], 'i').test(n.name),
+        );
+        if (matchingRegex >= 0) {
+          this.requiredFilesUnmatchedRegex.splice(matchingRegex, 1);
+        }
+      });
```

Comment:
I suggest this becomes a helper function called after any file is added or removed from the list. This should avoid the issue listed above.

Reply:
Alternatively, this could be computed below, where files are listed, with a local temporary variable, instead of keeping the list of unmatched regexes up-to-date.

Reply:
Good catch! I've updated the code and made this a helper function.